tests/components/test_component_manifold_extraction.cpp

```diff
@@ -0,0 +1,207 @@
+// #include <catch2/catch.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <wmtk_components/delaunay/internal/delaunay_2d.hpp>
+#include <wmtk_components/delaunay/internal/delaunay_3d.hpp>
+#include <wmtk_components/manifold_extraction/internal/Manifold-extraction.hpp>
+#include <paraviewo/VTUWriter.hpp>
+#include <igl/adjacency_matrix.h>
+#include <igl/connected_components.h>
+
+
+auto tagassign(size_t nb_triangles, double prob) -> std::vector<size_t>{
+    std::vector<size_t> tagass;
+    std::random_device rd{};
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<double> dis(0, 1);
+    // std::srand(10);
+    for (size_t i = 0 ; i < nb_triangles; i++){
+        double tag = dis(gen);
+        if (tag < prob) tagass.push_back(i); // for every tagged triangle, add its index to the end
+    }
+    return tagass;
+}
+
+TEST_CASE("Manifold-Extraction2D", "[components][man-ext2d]"){
+    
+    using namespace wmtk;
```

Comment:
This file is one of the rare cases (because it's a .cpp file) where it's totally fine to write this statement on the top of the file.